File: 283-MoveZeroes/283-MoveZeroes.java
// Last updated: 01/06/2026, 09:29:40
1class Solution {
2    public void moveZeroes(int[] nums) {
3        int j = 0;
4
5        for (int i = 0; i < nums.length; i++) {
6            if (nums[i] != 0) {
7                int temp = nums[i];
8                nums[i] = nums[j];
9                nums[j] = temp;
10                j++;
11            }
12        }        
13    }
14}

File: 283-MoveZeroes/283-MoveZeroes.py
# Last updated: 05/08/2026, 10:29:16
class Solution(object):
    def moveZeroes(self, nums):
        j = 0
        for i in range(0, len(nums)):
            if nums[i] != 0:
                t = nums[i]
                nums[i] = nums[j]
                nums[j] = t
                j += 1